pkg/mimirtool/commands/rules.go

```diff
@@ -407,7 +410,38 @@ func (r *RuleCommand) printRules(_ *kingpin.ParseContext) error {
 	}
 
 	p := printer.New(r.DisableColor)
-	return p.PrintRuleGroups(rules)
+
+	if r.OutputDir != "" {
+		log.Infof("Output dir detected writing rules to directory: %s", r.OutputDir)
+		for namespace, rule := range ruleNS {
+			if err = saveNamespaceRuleGroup(namespace, rule, r.OutputDir); err != nil {
+				return err
+			}
+		}
+
+		// Don't print the rule set if we've specified an output directory to save the rule files. It gets too noisy.
+		return nil
+	}
+
+	return p.PrintRuleGroups(ruleNS)
+}
+
+func saveNamespaceRuleGroup(ns string, ruleGroup []rwrulefmt.RuleGroup, dir string) error {
+	baseDir, err := filepath.Abs(dir)
+	if err != nil {
+		return err
+	}
+	file := filepath.Join(baseDir, fmt.Sprintf("%s.yaml", ns))
```

Comment:
I still think we should sanitize the namespace here, as we allow almost any characters in namespaces (eg. `/`), but we can do that later.

Reply:
I'm happy to do it now! What other cases do you have in mind besides `/` so that I can cover them? `filepath.Join(` calls [`filepath.Clean`](https://pkg.go.dev/path/filepath#Clean) on the result, which in my eyes would take care of the `/` and `.` problem.

I was thinking of something along the lines of:

```golang
filename := fmt.Sprintf("%s.yaml", ns) 
problematicChars := regexp.MustCompile(`[<>:"/\\|?*\x00-\x1F]`)
safeFilename := invalidChars.ReplaceAllString(filename, "_")
safeFilename = strings.TrimSpace(safeFilename)
```

WDYT?

Reply:
> What other cases do you have in mind besides / so that I can cover them?

Your suggestion with windows-forbidden chars makes sense to me.

> filepath.Join( calls [filepath.Clean](https://pkg.go.dev/path/filepath#Clean) on the result, which in my eyes would take care of the / and . problem.

`filepth.Join("dir", "a/b/c.yaml")` would still return `dir/a/b/c.yaml` and saving such file would fail becase we don't create subdirectories. But since `/` is now replaced with `_`, that solves the problem.